core/templates/dev/head/domain/collection/CollectionObjectFactory.js

```diff
@@ -192,6 +211,7 @@ oppia.factory('CollectionObjectFactory', [
     // skills from all collections nodes within this domain object. Please note
     // this operation returns a new skill list everytime this function is
     // called.
+    // TODO(wxy): migrate this to use the skills dict instead.
```

Comment:
yes, `SkillListObjectFactory`s are used for lists of skill ids in prereq/acquired skills